app/views/govuk_publishing_components/components/docs/search_with_autocomplete.yml

```diff
@@ -0,0 +1,61 @@
+name: Search with autocomplete (experimental)
+description: |
+  A version of the search component enhanced with the ability to fetch and display search
+  suggestions from a remote source as a user types.
+body: |
+  This component uses [Accessible Autocomplete](https://github.com/alphagov/accessible-autocomplete)
+  to enhance the [`search` component's](/component-guide/search) input element.
+  If the user does not have JavaScript enabled, the search component will function as normal.
+
+  The Accessible Autocomplete component is enhanced as follows:
+
+    * The input generated by the component is given the correct attributes and classes from the
+    original input field it is meant to enhance
+    * Suggestions are fetched from a remote source as the user types
+    * Parts of suggestions are highlighted where they match the user's input
+    * If the component is located in a form, the form is submitted
+    on selection of a suggestion through the component
+
+  Note that the inline label on the `search` component conflicts with the markup and styling
+  generated internally by Accessible Autocomplete. Our current designs do not foresee us using
+  autocomplete on a search box with an inline label, and this component will always force the
+  `inline_label` option on its nested `search` component to be `false`.
+
+  Note that this component is experimental and has not yet undergone AB testing and a thorough
+  accessibility audit. Until then, it should be used with caution and only and only by the GOV.UK
+  Search team.
+accessibility_criteria: |
+  This component should meet the accessibility acceptance criteria outlined in the [child search
+  component](/component-guide/search#accessibility-acceptance-criteria), as well as those of the
+  external [Accessibile Autocomplete
+  Component](https://github.com/alphagov/accessible-autocomplete/blob/master/accessibility-criteria.md)
+  project.
+examples:
+  default:
+    description: |
+      The default search component with autocomplete enabled.
+    data:
+      source: 'https://www.gov.uk/api/search.json?suggest=autocomplete'
+      source_key: suggested_autocomplete
```

Comment:
Probably worth using this document to explain what this option is for, I wasn't too sure without reading the code